Source/Bogus.Tests/DataSetTests/DateTest.cs

```diff
@@ -352,5 +358,267 @@ public void can_get_timezone_string()
       {
          date.TimeZoneString().Should().Be("Asia/Yerevan");
       }
+
+      public class FactWhenDaylightSavingsSupported : FactAttribute
+      {
+         public FactWhenDaylightSavingsSupported()
+         {
+            if (!TimeZoneInfo.Local.SupportsDaylightSavingTime)
+            {
+               Skip = "Test is only meaningful when Daylight Savings is supported by the local timezone.";
+            }
+         }
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void will_not_generate_values_that_do_not_exist_due_to_daylight_savings()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
+
+         // When converting back, .NET picks the end of the transition window instead of the start.
+         var transitionEndTime = transitionStartTime.ToUniversalTime().ToLocalTime();
+
+         // Act
+         var value = faker.Date.Between(transitionStartTime.AddHours(-1), transitionEndTime.AddHours(+2));
+
+         // Assert
+         transitionEndTime.Should().NotBe(transitionStartTime);
+
+         if ((value >= transitionStartTime) && (value < transitionStartTime.AddHours(1)))
+            value.Should().NotBeBefore(transitionEndTime);
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void will_adjust_start_time_to_avoid_dst_transition()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
+         var transitionEndTime = transitionStartTime + effectiveRule.DaylightDelta;
+
+         var windowStart = transitionStartTime + TimeSpan.FromTicks((transitionEndTime - transitionStartTime).Ticks / 2);
+         var windowEnd = transitionEndTime.AddMinutes(30);
+
+         // Act & Assert
+         bool haveSampleThatIsNotWindowEnd = false;
+
+         for (int i = 0; i < 10000; i++)
+         {
+            var sample = faker.Date.Between(windowStart, windowEnd);
+
+            sample.Should().BeOnOrAfter(transitionEndTime);
+            sample.Should().BeOnOrBefore(windowEnd);
+
+            haveSampleThatIsNotWindowEnd = (sample < windowEnd);
+         }
+
+         haveSampleThatIsNotWindowEnd.Should().BeTrue(because: "the effective range should include values other than windowEnd");
```

Comment:
I guess this is really nitpicky, so feel free to ignore this:
`windowEnd` in the string would not need to be hard-coded if using `nameof(windowEnd)` instead.

Reply:
I am in favour of this kind of thing :-)